source/agora/common/crypto/Schnorr.d

```diff
@@ -399,6 +399,69 @@ nothrow @nogc @safe unittest
     assert(stolen_key != kp.v);
 }
 
+/// Single-signer as part of multi-sig
+public Signature multiSigSign (const ref Pair R,
+    const ref Scalar v, const ref Scalar c) nothrow @nogc @safe
+{
+    Scalar s = R.v + (v * c);
+    return Sig(R.V, s).toBlob();
+}
```

Comment:
That's essentially doing `sign`, isn't it ? Except `c` is provided instead of being hashed.

Reply:
The `sign` is also including Point X (Public Key) and R (commitment) in the hash to create scalar c. For multisig this is a sum of X and R for all that can sign the block.
For normal Schnorr signatures this is required to prevent an attack where a key can be cancelled out by the attacker choosing rogue keys.
For block signing both X and R are already committed by the validators before the block hash is known so I think we do not need to include them in the challenge hash and it will not add to the security.